### ISSUE TITLE
docs(website): video links could be confusing to screen reader users

### ISSUE THREAD

**Issue by dawei-wang** [facebook/docusaurus#8465]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Both of the video links are announced as Explain like I'm 5 Docusaurus when a user tabs to them to them via screen reader. This can be confusing to screen reader users. Per WCAG 2.4.4 , the link text should be distinct because they lead to two different locations (YouTube videos in this case).
https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Operable

### Steps to reproduce

1. Start NVDA
2. Navigate to https://docusaurus.io/ 
3. Tab until you get to the video region
4. Both videos are announced as: Explain like I'm 5 Docusaurus

### Expected behavior

I expect a different title for the second video. The current behavior violates WCAG 2.4.4 and 2.4.6 

### Actual behavior

Both videos are announced as: Explain like I'm 5 Docusaurus

### Your environment

- Public source code:
- Public site URL: https://docusaurus.io/
- Docusaurus version used:  Canary
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Edge 108
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows 11

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Indeed, the second one should be called "a demo" or something similar. It was a copy/paste oversight. Welcoming PRs.
